r/R/make_pseudobulk_matrix.R

```diff
@@ -0,0 +1,41 @@
+#' Make aggregated matrix for pseudo bulk differential expression
+#'
+#' The SeuratObject requires a `custom` meta.data slot, specifying which cells
+#' belong to the `base` and `background` groups. It also needs a `samples`
+#' meta.data slot.
+#'
+#' @param scdata a SeuratObject,
+#'
+#' @return a SeuratObject with counts aggregated by sample for one cluster
+#' @export
+#'
+makePseudobulkMatrix <- function(scdata) {
+  counts <- scdata[["RNA"]]@counts
+  gene_annotations <- scdata@misc$gene_annotations
+
+  # create groups for aggregation
+  pbulk_groups <-
+    factor(paste(scdata$samples, scdata$custom, sep = "_"))
```

Comment:
This should use a better separator. samples usually contain underscores.

Reply:
once you subset scdata you can just do the aggregation on samples

Reply:
yeah! that was one of the things I wasn't sure about. that makes it simpler. I'll change it